x/tradeshield/keeper/keeper.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/elys-network/elys/x/tradeshield/types"
+)
+
+type (
+	Keeper struct {
+		cdc        codec.BinaryCodec
+		storeKey   storetypes.StoreKey
+		memKey     storetypes.StoreKey
+		paramstore paramtypes.Subspace
```

Comment:
We don't need paramSubspace. If params are required, they should be stored in the module KV Store. [Ref](https://github.com/cosmos/cosmos-sdk/tree/main/x/params)

Also, having in module KV store, upgrade to cosmos v0.50.0 will be easier

Reply:
I used ignite command,, does it mean we need to upgrade first ?

Reply:
also in reference those globally available params, see. this line [here](https://github.com/cosmos/cosmos-sdk/tree/main/x/params#abstract)

To use that we might have to keep all params there (leveragelp, perp, amm, etc)

Reply:
We can keep the params, but I recommend not using paramstaore subspace. It got deprecated more than 2 years ago

Note section in [Yes when cosmos was initially developed they were using ](Ref)

The getters and setters can manually be changed [leverageLP Getters](https://github.com/elys-network/elys/blob/main/x/leveragelp/keeper/params.go#L45) [leverageLP Setters](https://github.com/elys-network/elys/blob/main/x/leveragelp/keeper/params.go#L9)

Reply:
got it, thanks, going to update it

Reply:
done